contracts/CarbonVoteXCore.sol

```diff
@@ -359,4 +358,11 @@ contract CarbonVoteXCore is DSGuard {
         emit _SendGas(msg.sender, namespace, pollId, msg.value);
     }
 
+    // an internal function to initialize availablePermissions and restrictedFunctionSig
+    function initFunctionSig() internal {
+        availablePermissions.push(keccak256("register"));
```

Comment:
add state variables to avoid repeat typing, and replace all "register" and "sendGas":

string constant REGISTER = "register";
string constant SEND_GAS = "sendGas";